inference/core/workflows/execution_engine/dynamic_blocs/entities.py

```diff
@@ -0,0 +1,130 @@
+        "selectors.",
+    )
+    selector_data_kind: Dict[SelectorType, List[str]] = Field(
+        default_factory=dict,
+        description="Mapping of `selector_types` into names of kinds to be compatible. "
+        "Empty dict (default value) means wildcard kind for all selectors. If name of kind given - "
+        "must be valid kind, known for workflow execution engine.",
+    )
+    value_types: List[ValueType] = Field(
+        default_factory=list,
+        description="List of types representing union of types for static values (non selectors) "
+        "that shall be accepted for input field. Empty list represents no value types allowed.",
+    )
+
+
+class DynamicOutputDefinition(BaseModel):
+    type: Literal["DynamicOutputDefinition"]
+    kind: List[str] = Field(
+        default_factory=list,
+        description="List representing union of kinds for defined output",
+    )
+
+
+class ManifestDescription(BaseModel):
+    type: Literal["ManifestDescription"]
+    block_type: str = Field(
+        description="Field holds type of the bock to be dynamically created. Block can be initialised "
+        "as step using the type declared in the field."
+    )
+    inputs: Dict[str, DynamicInputDefinition] = Field(
+        description="Mapping name -> input definition for block inputs (parameters for run() function of"
+        "dynamic block)"
+    )
+    outputs: Dict[str, DynamicOutputDefinition] = Field(
+        default_factory=dict,
+        description="Mapping name -> output kind for block outputs.",
+    )
+    output_dimensionality_offset: int = Field(
+        default=0, ge=-1, le=1, description="Definition of output dimensionality offset"
+    )
+    accepts_batch_input: bool = Field(
+        default=False,
+        description="Flag to decide if function will be provided with batch data as whole or with singular "
+        "batch elements while execution",
+    )
+    accepts_empty_values: bool = Field(
+        default=False,
+        description="Flag to decide if empty (optional) values will be shipped as run() function parameters",
+    )
+
+
+class PythonCode(BaseModel):
+    type: Literal["PythonCode"]
+    function_code: str = Field(
+        description="Code of python function. Content should be properly formatted including indentations. "
+        "Workflows execution engine is to create dynamic module with provided function - ensuring "
+        "imports of the following symbols: [Any, List, Dict, Set, sv, np, math, Batch, "
+        "WorkflowImageData, BlockResult]"
+    )
+    function_name: str = Field(
+        default="run", description="Name of the function shipped in `function_code`."
+    )
+
+
+class DynamicBlockDefinition(BaseModel):
+    type: Literal["DynamicBlockDefinition"]
+    manifest: ManifestDescription = Field(
+        description="Definition of manifest for dynamic block to be created in runtime by "
+        "workflows execution engine."
+    )
+    code: PythonCode = Field(
```

Comment:
Probably also needs an init function

Reply:
👍 changed the way how it will work to have init function and run function with `self` injected as first param which would have access to `self._init_results`